### PR TITLE
enable boolean8 tests and call functions from the "omni" pkg

### PR DESCRIPTION
**Pull request by guilhermeleobas**
- explicit cast of boolean8 to bool
- call functions from omni. pkg rather than np.
- replace int8[] by bool[] in some tests

This PR requires a small change in omniscidb to handle boolean values: https://github.com/omnisci/omniscidb-internal/pull/4689

### CODE REVIEW

**Comment by guilhermeleobas** [comment]
PR is ready for review. Wait for omnisci/omniscidb-internal/pull/4689 to be merged before merging this one.

File: rbc/omnisci_backend/numpy_ufuncs.py [hunk 1]
```diff
@@ -9,7 +9,17 @@
 
 
 def determine_dtype(a, dtype):
-    return a.eltype if dtype is None else dtype
+    if isinstance(a, ArrayPointer):
+        return a.eltype if dtype is None else dtype
+    else:
+        return a if dtype is None else dtype
```
Comment (pearu) [inline rbc/omnisci_backend/numpy_ufuncs.py:15]: `return a` is suspicious: the function ought to return type instance..

File: rbc/omnisci_backend/numpy_ufuncs.py [hunk 2]
```diff
@@ -9,7 +9,17 @@
...
+
+
+def determine_input_type(argty):
+    if argty == typesystem.boolean8:
+        return bool
+    else:
+        return argty.eltype if isinstance(argty, ArrayPointer) else argty
```
Comment (pearu) [inline rbc/omnisci_backend/numpy_ufuncs.py:22]: What are the allowed return values of `determine_dtype` and `determine_input_type`? Python types, Numba types, anything else? I presume to return typesystem types is not allowed.

File: rbc/tests/test_omnisci_math.py [hunk 3]
```diff
@@ -162,33 +164,43 @@ def omnisci():
...
+        fn.__name__ = fn_name
 
-    if fn_name in ['logical_or', 'logical_xor', 'logical_and', 'logical_not']:
+    if available_version[:2] <= (5, 3) and fn_name in \
+            ['logical_or', 'logical_xor', 'logical_and', 'logical_not']:
         # Invalid use of Function(<ufunc 'logical_or'>) with
         # argument(s) of type(s): (boolean8, boolean8)
-        pytest.skip(f'using boolean8 as {fn_name} argument not implemented')
+        pytest.skip(
+            f'using boolean8 as {fn_name} argument not implemented for'
```
Comment (pearu) [inline rbc/tests/test_omnisci_math.py:195]: suggest rewording: `"using boolean arguments requires omniscidb v 5.4 or newer (got {available_version})"`

File: rbc/omnisci_backend/numpy_ufuncs.py (hunk 2 above)
Comment (guilhermeleobas) [reply rbc/omnisci_backend/numpy_ufuncs.py:22]: `determine_dtype` and `determine_input_type` must return numba types.

* `determine_input_type`: Only exists to explicitly convert `boolean8` values to `bool`. Without this, one would have to write numpy functions to work with `boolean8` args (see #146). 
* `determine_dtype`: Explicit convert the output to a dtype.

> I presume to return typesystem types is not allowed.

Yes, you're right.

File: rbc/omnisci_backend/numpy_ufuncs.py (hunk 1 above)
Comment (guilhermeleobas) [reply rbc/omnisci_backend/numpy_ufuncs.py:15]: `a` ought to be a type. I can add a check at the beginning of this function to make sure it is a type. What do you think?

Comment (pearu) [reply rbc/omnisci_backend/numpy_ufuncs.py:15]: Perhaps introduce a decorator on  determine_dtype and determine_input_type that checks that the return type is numba type?

Comment (pearu) [reply rbc/omnisci_backend/numpy_ufuncs.py:15]: Can you point the case where `return a` is effective?

Comment (guilhermeleobas) [reply rbc/omnisci_backend/numpy_ufuncs.py:15]: Yes. Look at the implementation of `np.negative(<int64>)` for instance. In this case `return a` returns `int64` since the dtype is `None`. But on `np.logical_not`, the output needs to be a boolean value, and we enforce this constraint using the dtype argument.
